flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2116,7 +2116,9 @@ class ArrayExprLowering {
           loc, zero, i.value(), one, /*unordered=*/false,
           /*finalCount=*/false, mlir::ValueRange{innerArg});
       innerArg = loop.getRegionIterArgs().front();
-      ivars.push_back(loop.getInductionVar());
+      // Store induction vars in column major order, which is what FIR array ops
+      // expect.
+      ivars[loopDepth - 1 - i.index()] = loop.getInductionVar();
```

Comment:
Thanks for the fix.

I think in other spots that I built the list backwards and then just reversed it. Forgot the reverse call here.